text/src/tr.rs

```diff
@@ -0,0 +1,687 @@
+            return Err("Need two strings operand".to_string());
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Char {
+    char: char,
+    repeated: usize,
+}
+
+#[derive(Debug, Clone)]
+struct Equiv {
+    char: char,
+}
+
+#[derive(Debug, Clone)]
+enum Operand {
+    Char(Char),
+    Equiv(Equiv),
+}
+
+impl Operand {
+    fn contains(operands: &Vec<Operand>, target: &char) -> bool {
+        for operand in operands {
+            match operand {
+                Operand::Equiv(e) => {
+                    if compare_deunicoded_chars(e.char, *target) {
+                        return true;
+                    }
+                }
+                Operand::Char(c) => {
+                    if c.char == *target {
+                        return true;
+                    }
+                }
+            }
+        }
+        false
+    }
+}
+
+fn filter_chars(operands: Vec<Operand>) -> Vec<Char> {
+    operands
+        .into_iter()
+        .filter_map(|operand| {
+            if let Operand::Char(c) = operand {
+                Some(c)
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
+fn create_minimal_string(chars: Vec<Char>, size: usize) -> Vec<char> {
+    let mut result = vec![];
+    let mut remaining_space = size;
+    let mut overflow_chars: Vec<(usize, Char)> = vec![];
+
+    // Add chars with repeated == 1 to the result
+    for ch in &chars {
+        if ch.repeated == 1 {
+            if remaining_space > 0 {
+                result.push(ch.char);
+                remaining_space -= 1;
+            }
+        } else if remaining_space >= ch.repeated {
+            for _ in 0..ch.repeated {
+                result.push(ch.char);
+            }
+
+            remaining_space -= ch.repeated;
+        } else {
+            overflow_chars.push((result.len(), ch.clone()));
+        }
+    }
+
+    // Add remaining chars from overflow_chars if there's still space
+    if !overflow_chars.is_empty() {
+        for (insert_position, char) in overflow_chars.iter().rev() {
+            if remaining_space > 0 {
+                let chars_to_add = remaining_space.min(char.repeated);
+                let replace_with = vec![char.char; chars_to_add];
+                result.splice(insert_position..insert_position, replace_with);
+
+                remaining_space -= chars_to_add;
+            }
+        }
+    }
+
+    if result.len() < size {
+        let last = *result.last().unwrap();
+        for _ in 0..size - result.len() {
+            result.push(last);
+        }
+    }
+
+    result
+}
+
+fn parse_symbols(input: &str) -> Result<Vec<Operand>, String> {
+    let mut operands: Vec<Operand> = Vec::new();
+    let mut chars = input.chars().peekable();
+
+    while let Some(&ch) = chars.peek() {
+        if ch == '[' {
+            chars.next(); // Skip '['
+            if let Some(&'=') = chars.peek() {
```

Comment:
Flatten a bit. You could utilize `let-else` pattern. Not exactly for this case, but other `if let`s could be refactored like that. Even better would be to use `.ok_or` method on options.